test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp

```diff
@@ -162,8 +206,8 @@ class ImageInventoryTest : public CppUnit::TestFixture
 		for (unsigned i = 0; i < images.size(); i++)
 		{
 			bool flag = false;
-			int rc = mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, (sizeof(currentId)/sizeof(currentId[0])) - 1);
-			if (rc > 0 && rc < sizeof(currentId)/sizeof(currentId[0])) currentId[rc] = 0;
+			int rc = (int)mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, (sizeof(currentId)/sizeof(currentId[0])) - 1);
```

Comment:
But it probably doesn't use some "random negative value", it probably uses some specific value (like 0xffffffff) or something. Up, it says it returns a (size_t) -1, test for that, and type RC properly, thanks.